pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -305,6 +305,7 @@ public void deleteSegmentFile() {
 
   private final StreamPartitionMsgOffset _latestStreamOffsetAtStartupTime;
   private final CompletionMode _segmentCompletionMode;
+  private final List<String> _filteredMessageOffsets = new ArrayList<>();
```

Comment:
I wouldn't really track the individual offsets as it can easily flood the log. I know there are use cases where most of the records are filtered out because multiple tables share the same stream. Do you really see the offset info useful?

Reply:
> I wouldn't really track the individual offsets as it can easily flood the log. 

We are not tracking individual offsets but logging once a minute just like consumed events log. 

>  Do you really see the offset info useful?

So we have a usecase where we don't want to ingest an invalid json in our table. For that we are planning to use isJson UDF as a filterConfig in #12603. Now we will have an alert on the metric `NUMBER_ROWS_FILTERED` and having a log to track the offset would really help in debugging faster.

But open to suggestions if we can do it in a better way. Maybe having a config in FilterConfig to log / track and we enable it only for tables we really need to in case of debugging?

Reply:
What I meant is that we can have a counter, but logging all offsets might cause problems. It might end up logging thousands of offsets per minute easily. I'm okay controlling this with a debug flag

Reply:
> I'm okay controlling this with a debug flag

@Jackie-Jiang just want to confirm on the implementation details.. is it okay to add it as a config `trackFilteredRowOffsets` in [IngestionConfig class](https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/config/table/ingestion/IngestionConfig.java#L65 )?

Reply:
I feel it is more suitable under the `StreamIngestionConfig` because offset only applies to streaming ingestion. How about `trackFilteredMessageOffsets`?

Reply:
+1 to add that as a config with the default value being false.